store/memory.go

```diff
@@ -0,0 +1,145 @@
+package store
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	MemStoreFlagName   = "memstore.enabled"
+	ExpirationFlagName = "memstore.expiration"
+
+	DefaultPruneInterval = 500 * time.Millisecond
+)
+
+type MemStoreConfig struct {
+	Enabled        bool
+	BlobExpiration time.Duration
+}
+
+// MemStore is a simple in-memory store for blobs which uses an expiration
+// time to evict blobs to best emulate the ephemeral nature of blobs dispersed to
+// EigenDA operators.
+type MemStore struct {
+	sync.RWMutex
+
+	cfg       *MemStoreConfig
+	keyStarts map[string]time.Time
+	store     map[string][]byte
+}
+
+// NewMemStore ... constructor
+func NewMemStore(ctx context.Context, cfg *MemStoreConfig) (*MemStore, error) {
+	store := &MemStore{
+		cfg:       cfg,
+		keyStarts: make(map[string]time.Time),
+		store:     make(map[string][]byte),
+	}
+
+	if cfg.BlobExpiration != 0 {
+		go store.EventLoop(ctx)
+	}
+
+	return store, nil
+}
+
+func (e *MemStore) EventLoop(ctx context.Context) {
+
+	timer := time.NewTicker(DefaultPruneInterval)
+
+	select {
+	case <-ctx.Done():
+		return
+
+	case <-timer.C:
+		e.pruneExpired()
+	}
+
+}
+
+func (e *MemStore) pruneExpired() {
+	e.Lock()
+	defer e.Unlock()
+
+	for commit, dur := range e.keyStarts {
+		if time.Since(dur) >= e.cfg.BlobExpiration {
+			delete(e.keyStarts, commit)
+			delete(e.store, commit)
+		}
+	}
+
+}
+
+// Get fetches a value from the store.
+func (e *MemStore) Get(ctx context.Context, commit []byte) ([]byte, error) {
+	e.RLock()
+	defer e.RUnlock()
+
+	key := common.Bytes2Hex(commit)
+	if _, exists := e.store[key]; !exists {
+		return nil, fmt.Errorf("commitment key not found")
+	}
+
+	return e.store[key], nil
+}
+
+// Put inserts a value into the store.
+func (e *MemStore) Put(ctx context.Context, value []byte) ([]byte, error) {
+	e.Lock()
```

Comment:
Is a map in golang not threadsafe? Is there a need to lock here?

Reply:
After some googling, apparently it is not threadsafe for writes, unless you use sync.Map. But this seems fine for now.